File: 2025-12/src/day_03/boj_1735/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int a, b, c, d;
    cin >> a >> b >> c >> d;

    int numerator = a * d + b * c;  // 분자
    int denominator = b * d;        // 분모
    int g = gcd(numerator, denominator);

    cout << numerator / g << ' ' << denominator / g;
}

File: 2025-12/src/day_03/boj_1735/Main.java
package day_03.boj_1735;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        st = new StringTokenizer(br.readLine());
        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());
        int c = Integer.parseInt(st.nextToken());
        int d = Integer.parseInt(st.nextToken());

        int numerator = a * d + b * c;  // 분자
        int denominator = b * d;  // 분모
        int gcd = gcd(numerator, denominator);

        System.out.printf("%d %d", numerator / gcd, denominator / gcd);
    }

    static int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }
}
